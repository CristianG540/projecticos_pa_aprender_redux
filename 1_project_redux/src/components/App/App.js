import React, { Component } from 'react'
import HelloWorld from '../HelloWorld/HelloWorld'

class App extends Component {
  caca = 'hehehe'

  render () {
    return <HelloWorld tech={this.caca} />
  }
}

export default App
