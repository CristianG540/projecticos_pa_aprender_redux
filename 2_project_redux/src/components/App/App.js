import React, { Component } from 'react'

// Redux
import store from '../../redux/store'

// Components
import HelloWorld from '../HelloWorld/HelloWorld'
import ButtonGroup from '../ButtonGroup/ButtonGroup'

class App extends Component {
  render () {
    return (
      <>
        <HelloWorld tech={store.getState().tech} />
        <ButtonGroup technologies={['React', 'Elm', 'React-redux']} />
      </>
    )
  }
}

export default App
