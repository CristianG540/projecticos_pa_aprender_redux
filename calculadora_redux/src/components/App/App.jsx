import React, { Component } from 'react'
// Redux
import store from '../../redux/store'
// Components
import CalculatorButton from '../CalculatorButton/CalculatorButton'
// Constants
import { ALL_CLEAR } from '../CalculatorButton/constants'

class App extends Component {
  render () {
    return (
      <div className='calculator'>
        <textarea className='calculator-screen' value={store.getState().screen.content} disabled />
        <div className='calculator-keys'>
          <CalculatorButton id='plus-btn' text='+' cssClass='operator' />
          <CalculatorButton id='minus-btn' text='-' cssClass='operator' />
          <CalculatorButton id='multiply-btn' text='*' cssClass='operator' />
          <CalculatorButton id='divide-btn' text='/' cssClass='operator' />

          <CalculatorButton id='seven-btn' text='7' />
          <CalculatorButton id='eight-btn' text='8' />
          <CalculatorButton id='nine-btn' text='9' />

          <CalculatorButton id='four-btn' text='4' />
          <CalculatorButton id='five-btn' text='5' />
          <CalculatorButton id='six-btn' text='6' />

          <CalculatorButton id='one-btn' text='1' />
          <CalculatorButton id='two-btn' text='2' />
          <CalculatorButton id='three-btn' text='3' />

          <CalculatorButton id='zero-btn' text='0' />
          <CalculatorButton id='decimal-btn' text='.' cssClass='decimal' />
          <CalculatorButton id='all-clear-btn' text='AC' value={ALL_CLEAR} cssClass='all-clear' />
          <CalculatorButton id='equal-btn' text='=' cssClass='equal-sign' />
        </div>
      </div>
    )
  }
}

export default App
