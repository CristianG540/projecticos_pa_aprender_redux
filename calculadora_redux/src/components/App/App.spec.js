import React from 'react'
import { render } from '@testing-library/react'
import App from './App'

describe('App component', () => {
  it('should take a snapshot', () => {
    const { asFragment } = render(<App />)

    expect(asFragment(<App />)).toMatchSnapshot()
  })
})
