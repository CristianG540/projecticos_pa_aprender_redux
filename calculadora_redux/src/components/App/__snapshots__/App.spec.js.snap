// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="calculator"
  >
    <textarea
      class="calculator-screen"
      disabled=""
    >
      0
    </textarea>
    <div
      class="calculator-keys"
    >
      <button
        class="operator"
        id="plus-btn"
        value="+"
      >
        +
      </button>
      <button
        class="operator"
        id="minus-btn"
        value="-"
      >
        -
      </button>
      <button
        class="operator"
        id="multiply-btn"
        value="*"
      >
        *
      </button>
      <button
        class="operator"
        id="divide-btn"
        value="/"
      >
        /
      </button>
      <button
        class="[]"
        id="seven-btn"
        value="7"
      >
        7
      </button>
      <button
        class="[]"
        id="eight-btn"
        value="8"
      >
        8
      </button>
      <button
        class="[]"
        id="nine-btn"
        value="9"
      >
        9
      </button>
      <button
        class="[]"
        id="four-btn"
        value="4"
      >
        4
      </button>
      <button
        class="[]"
        id="five-btn"
        value="5"
      >
        5
      </button>
      <button
        class="[]"
        id="six-btn"
        value="6"
      >
        6
      </button>
      <button
        class="[]"
        id="one-btn"
        value="1"
      >
        1
      </button>
      <button
        class="[]"
        id="two-btn"
        value="2"
      >
        2
      </button>
      <button
        class="[]"
        id="three-btn"
        value="3"
      >
        3
      </button>
      <button
        class="[]"
        id="zero-btn"
        value="0"
      >
        0
      </button>
      <button
        class="decimal"
        id="decimal-btn"
        value="."
      >
        .
      </button>
      <button
        class="all-clear"
        id="all-clear-btn"
        value="all-clear"
      >
        AC
      </button>
      <button
        class="equal-sign"
        id="equal-btn"
        value="="
      >
        =
      </button>
    </div>
  </div>
</DocumentFragment>
`;
